ui/dashboards/src/components/Variables/variable-model.ts

```diff
@@ -0,0 +1,68 @@
+// Copyright 2022 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { ListVariableDefinition } from '@perses-dev/core';
+import {
+  usePlugin,
+  useTemplateVariableValues,
+  useDatasourceStore,
+  useTimeRange,
+  VariableStateMap,
+} from '@perses-dev/plugin-system';
+import { useQuery } from '@tanstack/react-query';
+
+export function useListVariablePluginValues(definition: ListVariableDefinition) {
```

Comment:
This was refactored out of the <ListVariable /> component to be a standalone re-usable hook so it can be used in the edit form as well. This may even be better to put in the `plugin-system` package.